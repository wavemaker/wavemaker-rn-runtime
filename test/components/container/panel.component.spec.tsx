import React, { ReactNode } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import WmPanel from '@wavemaker/app-rn-runtime/components/container/panel/panel.component';

describe('Test Panel component', () => {
    test('Check validity of sample component', () => {
      const tree = renderer.create(<WmPanel name="test_Panel"/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
});