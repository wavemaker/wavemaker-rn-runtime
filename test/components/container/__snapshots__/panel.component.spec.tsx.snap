// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Panel component Check validity of sample component 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 0,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <View
    accessibilityRole="header"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={false}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "borderStyle": "solid",
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
        "borderWidth": 0,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
    testID="test_Panel_header"
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={false}
        style={
          [
            {
              "color": "#333333",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontWeight": "bold",
              "userSelect": "text",
            },
            {
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="test_Panel_header_title"
      >
        Title
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    />
  </View>
  <View
    animatedStyle={
      {
        "value": {
          "maxHeight": 0,
        },
      }
    }
    collapsable={false}
    style={
      {
        "maxHeight": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View />
    </View>
  </View>
</View>
`;
